hubbard/grid.py

```diff
@@ -0,0 +1,64 @@
+import numpy as np
+import sisl
+
+
+def real_space_grid(geometry, vector, shape, xmin, xmax, ymin, ymax, z, mode='wavefunction', **kwargs):
```

Comment:
The interface here forces it to be 2d with z being the dependent variable. Is that what you want for the future API? In case you want to handle other systems where this is not the natural order, then it might be worthwhile generalizing the interface?

Reply:
Hmm maybe I'm mistaken but I think that one can also create a 3D grid with the appropriate `shape`. The `z` variable sets only the origo which is handy if one wants to create a 2D slice of the grid at that particular height.

Reply:
Yes, but my point is that the plot forces a certain direction, not that I have anything against it, but in case you want to generalize this later, it is better to do it now since the API would change.

Reply:
OK understood your point, sorry. You are right. I have an idea, will push it in a few minutes :)